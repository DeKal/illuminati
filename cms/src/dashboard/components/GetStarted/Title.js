import { withStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'

const Title = withStyles(theme => ({
  root: {
    fontSize: theme.fontSizes.XXL,
    fontWeight: 600,
    padding: '9px 0 4px 0',
    lineHeight: 1.3
  }
}))(Typography)

export default Title
