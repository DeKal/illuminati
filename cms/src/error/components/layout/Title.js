import { styled } from '@material-ui/core/styles'

const Title = styled('h2')(({ theme }) => ({
  color: theme.colors.dark,
  fontSize: theme.fontSizes.XXXL,
  fontWeight: 700,
  textTransform: 'uppercase',
  margin: `${theme.spaces.xsmall}px`
}))

export default Title
