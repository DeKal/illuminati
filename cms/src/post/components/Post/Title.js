import React from 'react'
import PropTypes from 'prop-types'
import TextField from 'post/components/TextField'

const Title = ({ isEdited, title, editPostTitle }) => (
  <TextField
    label="Post title"
    required
    variant="outlined"
    value={title}
    multiline
    rowsMax={2}
    fullWidth
    error={isEdited}
    onChange={event => editPostTitle(event.target.value)}
  />
)

export default Title

Title.propTypes = {
  isEdited: PropTypes.bool,
  title: PropTypes.string,
  editPostTitle: PropTypes.func
}
