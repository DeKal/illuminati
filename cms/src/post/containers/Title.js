import { connect } from 'react-redux'
import { getPostTitle, getEditedTitle } from 'post/state/selectors'
import { editPostTitle } from 'post/state/actions'
import Title from 'post/components/Post/Title'

const mapStateToProps = state => {
  const editedTitle = getEditedTitle(state)
  const title = getPostTitle(state)
  return {
    title: editedTitle || '',
    isEdited: editedTitle !== title
  }
}

const mapDispatchToProps = {
  editPostTitle
}

export default connect(mapStateToProps, mapDispatchToProps)(Title)
