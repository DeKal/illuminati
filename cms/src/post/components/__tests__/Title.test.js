import * as React from 'react'
import { shallow } from 'enzyme'
import Title from '../Title'

describe('Title', () => {
  it('should render Title correctly', function() {
    const wrap = shallow(<Title title="title" />)
    expect(wrap).toMatchSnapshot()
  })
})
