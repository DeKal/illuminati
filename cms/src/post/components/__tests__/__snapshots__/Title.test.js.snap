// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render Title correctly 1`] = `
<WithStyles(WithStyles(ForwardRef(TextField)))
  fullWidth={true}
  label="Post title"
  multiline={true}
  onChange={[Function]}
  required={true}
  rowsMax={2}
  value="title"
  variant="outlined"
/>
`;
